backend/src/common/types/controller/controller.ts

```diff
@@ -18,9 +28,28 @@ interface TypedRequest<T extends Query, U> extends Request {
   query: T;
 }
 
+interface AuthRequest extends Request {
+  tokenPayload: TokenPayload;
+}
+
+type AuthTypedRequestBody<T> = TypedRequestBody<T> & AuthRequest;
+
+type AuthTypedRequestQuery<T extends Query> = TypedRequestQuery<T> &
+  AuthRequest;
+
+type AuthTypedRequestParams<T extends ParamsDictionary> =
+  TypedRequestParams<T> & AuthRequest;
+
+type AuthTypedRequest<T extends Query, U> = TypedRequest<T, U> & AuthRequest;
```

Comment:
Are these types still needed? Can't we just use the `TypedRequest...` as the `Request` already contains the `tokenPayload`

Reply:
But... then what did you mean by this?  
https://github.com/BinaryStudioAcademy/bsa-2022-autoline/pull/218#discussion_r964528131

Reply:
I meant that we must type the controller requests. But since then you moved `tokenPayload` to the global `Request` and we don't need to have additional `AuthTyped...` types anymore. Because of the existing `TypedRequest...` extends the `Request`